react/features/base/participants/middleware.js

```diff
@@ -407,9 +407,9 @@ function _participantJoinedOrUpdated(store, next, action) {
     // to the new avatar and emit out change events if necessary.
     const result = next(action);
 
-    const { disableThirdPartyRequests } = getState()['features/base/config'];
+    const { disableThirdPartyRequests, gravatarBaseURL } = getState()['features/base/config'];
 
-    if (!disableThirdPartyRequests && (avatarURL || email || id || name)) {
+    if ((!disableThirdPartyRequests || gravatarBaseURL) && (avatarURL || email || id || name)) {
```

Comment:
I believe we have to adapt these conditions if we allow an URL to match the origin if disableThirdPartyRequests is true

Reply:
Yeah, it's probably a good idea to encapsulate that in some helper function not to make this check more complex.

Reply:
Or would be it be acceptable to only check: `if (avatarURL || email || id || name)` ? since the avatar url check is done deeper in `_getFirstLoadableAvatarUrl`

Reply:
Probably yeah.